--- truecrypt-4.2a.a/Linux/Kernel/Dm-target.c	2007-03-07 15:04:47.000000000 +0100
+++ truecrypt-4.2a/Linux/Kernel/Dm-target.c	2007-03-08 01:48:38.000000000 +0100
@@ -15,7 +15,6 @@
 #include <linux/moduleparam.h>
 #include <linux/version.h>
 #include <linux/workqueue.h>
-#include <dm.h>
 
 #include "Tcdefs.h"
 #include "Crypto.h"
@@ -30,6 +29,27 @@
 #define DM_MSG_PREFIX "truecrypt"
 #endif
 
+#define DM_NAME "device-mapper"
+
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,18)
+#define DMERR(f, arg...) printk(KERN_ERR DM_NAME ": " DM_MSG_PREFIX ": " f "\n", ## arg)
+#else
+#define DMERR(f, x...) printk(KERN_ERR DM_NAME ": " f "\n" , ## x)
+#endif
+
+/* 
+ * struct dm_dev taken from 2.6.{17,18,19,20} drivers/md/dm.h 
+ */
+
+struct dm_dev {
+        struct list_head list;
+
+        atomic_t count;
+        int mode;
+        struct block_device *bdev;
+        char name[16];
+};
+
 int trace_level = 0;
 
 #define MSG_PREFIX "truecrypt: "
